TCD1304.py

```diff
@@ -62,29 +58,32 @@ def power_source ():        # puts a voltage of 4V on PV1
         ps.pv1 = 4
         ps.pv1
     
+    def analog_signal_read (self, block:False):
+        self.scope = Oscilloscope()

```

Comment:
I recommend creating `self.scope` in `__init__` instead of here. Otherwise a new scope will be created every time this method is called.